### ISSUE TITLE
Secret callback

### ISSUE THREAD

**Issue by andizzle** [reddit/baseplate.go#16]
When fetched secret is used for creating a downstream client, it's possible that we need to create a new client once the secret is updated. Add capability of being notified of updated secret via secret store will allow us to renew a client as needed.

Maybe something like 
```go
func NewStore(ctx context.Context, path string, logger log.Wrapper, notify chan interface) (*Store, error) {
	parser := func(r io.Reader) (interface{}, error) {
		secrets, err := NewSecrets(r)
		if err != nil {
			return nil, err
		}
                notify <- secrets
		return secrets, nil
	}
        ...
```

Also open for other suggestions.

**Comment by fishy** [comment]
I like the channel idea. Just two things to note:

1. We shouldn't change `NewStore` because most of the use cases doesn't need the callback. We can create a new one like `NewStoreWithCallback`
2. The `notify <- secrets` should probably be in its own goroutine, otherwise if the caller didn't read from the channel timely, this would block the parser.

**Comment by konradreiche** [comment]
What does a downstream client mean? Is that a client implementation you are providing yourself outside of baseplate.go?

**Comment by fishy** [comment]
> What does a downstream client mean? Is that a client implementation you are providing yourself outside of baseplate.go?

Yes.

**Comment by andizzle** [comment]
This is addressed by #20

**Comment by fishy** [comment]
We still need to allow registering of middlewares after initialization, but that can be in its own ticket.
